### PR TITLE
NIFI-11161 Simplify KeyedCipherPropertyEncryptor

### PR DESCRIPTION
**Pull request by exceptionfactory**
# Summary

[NIFI-11161](https://issues.apache.org/jira/browse/NIFI-11161) Simplifies the `KeyedCipherPropertyEncryptor` implementation that supports encryption and decryption of sensitive flow property values.

The existing implementation delegates `Cipher` instantiation to the `AESKeyedCipherProvider`, which includes a number of checks that are not necessary for the `KeyedCipherPropertyEncryptor`.

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [X] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [X] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [X] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [X] Pull Request based on current revision of the `main` branch
- [X] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [X] Build completed using `mvn clean install -P contrib-check`
  - [X] JDK 8
  - [ ] JDK 11
  - [ ] JDK 17

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

File: nifi-commons/nifi-property-encryptor/src/main/java/org/apache/nifi/encrypt/KeyedCipherPropertyEncryptor.java [hunk 1]
```diff
@@ -16,47 +16,36 @@
...
-    private static final boolean ENCRYPT = true;
+    private static final int GCM_TAG_LENGTH_BITS = 128;
 
-    private static final boolean DECRYPT = false;
-
-    private final KeyedCipherProvider cipherProvider;
+    private static final int ARRAY_START = 0;
 
-    private final EncryptionMethod encryptionMethod;
+    private static final String CIPHER_ALGORITHM = "AES/GCM/NoPadding";
```
Comment (greyp9) [inline nifi-commons/nifi-property-encryptor/src/main/java/org/apache/nifi/encrypt/KeyedCipherPropertyEncryptor.java:36]: The IV length, tag length, and algorithm are each declared multiple times in the code base.  Do you envision consolidating those usages at some point in the future?

File: nifi-commons/nifi-property-encryptor/src/test/java/org/apache/nifi/encrypt/KeyedCipherPropertyEncryptorTest.java [hunk 2]
```diff
@@ -77,26 +70,26 @@ public void testEncryptHexadecimalEncoded() throws DecoderException {
     @Test
     public void testDecryptEncryptionException() {
         final String encodedProperty = Hex.encodeHexString(PROPERTY.getBytes(DEFAULT_CHARSET));
-        assertThrows(EncryptionException.class, () -> encryptor.decrypt(encodedProperty));
+        assertThrows(Exception.class, () -> encryptor.decrypt(encodedProperty));
```
Comment (greyp9) [inline nifi-commons/nifi-property-encryptor/src/test/java/org/apache/nifi/encrypt/KeyedCipherPropertyEncryptorTest.java:73]: Why not retain the more specific exception type here?

**Review by greyp9 — COMMENTED** [review]
Nice refactor!  Couple of questions for follow-up...

File: nifi-commons/nifi-property-encryptor/src/main/java/org/apache/nifi/encrypt/KeyedCipherPropertyEncryptor.java [hunk 3]
```diff
@@ -16,47 +16,36 @@
...
     private final SecureRandom secureRandom;
 
     private final String description;
 
-    protected KeyedCipherPropertyEncryptor(final KeyedCipherProvider cipherProvider,
-                                           final EncryptionMethod encryptionMethod,
-                                           final SecretKey secretKey) {
-        this.cipherProvider = cipherProvider;
-        this.encryptionMethod = encryptionMethod;
+    protected KeyedCipherPropertyEncryptor(final SecretKey secretKey) {
```
Comment (greyp9) [inline nifi-commons/nifi-property-encryptor/src/main/java/org/apache/nifi/encrypt/KeyedCipherPropertyEncryptor.java:44]: Is `AES/GCM/NoPadding` the previous default?  Curious about the upgrade implications.

File: nifi-commons/nifi-property-encryptor/src/main/java/org/apache/nifi/encrypt/KeyedCipherPropertyEncryptor.java (hunk 1 above)
Comment (exceptionfactory) [reply nifi-commons/nifi-property-encryptor/src/main/java/org/apache/nifi/encrypt/KeyedCipherPropertyEncryptor.java:36]: That's a good question. Theoretically, all the values should be the same across the board, but we would need to implement this in a way that avoid introducing unnecessary dependencies. There is other work to do for removing support for deprecated legacy algorithms. That might be an opportunity for consolidation.

File: nifi-commons/nifi-property-encryptor/src/test/java/org/apache/nifi/encrypt/KeyedCipherPropertyEncryptorTest.java (hunk 2 above)
Comment (exceptionfactory) [reply nifi-commons/nifi-property-encryptor/src/test/java/org/apache/nifi/encrypt/KeyedCipherPropertyEncryptorTest.java:73]: The thrown exception is different based the Java version, so switching to the more generic Exception provided the least amount of impact to the unit test.

File: nifi-commons/nifi-property-encryptor/src/main/java/org/apache/nifi/encrypt/KeyedCipherPropertyEncryptor.java (hunk 3 above)
Comment (exceptionfactory) [reply nifi-commons/nifi-property-encryptor/src/main/java/org/apache/nifi/encrypt/KeyedCipherPropertyEncryptor.java:44]: Yes, `AES/GCM/NoPadding` is the current value for all instances of the `KeyedCipherProvider`, so this change does not impact compatibility with existing AES-GCM sensitive property algorithms.

**Review by greyp9 — APPROVED** [review]
LGTM, thanks!

**Pull request closed without merging**